File: 01-january/21-minimum-bitwise-array/solution.py
class Solution:
    def minBitwiseArray(self, nums):
        res = []
        for p in nums:
            if p % 2 == 0:
                res.append(-1)
                continue
            temp = p
            cnt = 0
            while temp & 1:
                cnt += 1
                temp >>= 1
            res.append(p - (1 << (cnt - 1)))
        return res

File: 01-january/21-minimum-bitwise-array/solution.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var minBitwiseArray = function(nums) {
  const res = new Array(nums.length);
  for (let i = 0; i < nums.length; i++) {
    const p = nums[i];
    if ((p & 1) === 0) {
      res[i] = -1;
      continue;
    }
    let temp = p;
    let cnt = 0;
    while ((temp & 1) === 1) {
      cnt++;
      temp >>= 1;
    }
    res[i] = p - (1 << (cnt - 1));
  }
  return res;
};
